core/test/Voting.js

```diff
@@ -1012,4 +1014,141 @@ contract("Voting", function(accounts) {
     result = await voting.retrieveRewards({ from: account4 });
     truffleAssert.eventNotEmitted(result, "RewardsRetrieved");
   });
+
+  it("Commit and persist encrypted price in a single transaction", async function() {
+    const identifier = web3.utils.utf8ToHex("commit-and-persist");
+    const time = "1000";
+    await voting.addSupportedIdentifier(identifier);
+
+    await voting.requestPrice(identifier, time, { from: registeredDerivative });
+    await moveToNextRound(voting);
+
+    const price = getRandomSignedInt();
+    const salt = getRandomUnsignedInt();
+    const hash = web3.utils.soliditySha3(price, salt);
+    const roundId = await voting.getCurrentRoundId();
+
+    const { privateKey, publicKey } = await deriveKeyPairFromSignature(web3, getKeyGenMessage(roundId), account1);
+    const vote = { price: price.toString(), salt: salt.toString() };
+    const encryptedMessage = await encryptMessage(publicKey, JSON.stringify(vote));
+
+    await voting.commitAndPersistEncryptedVote(identifier, time, hash, encryptedMessage);
+
+    const topicHash = computeTopicHash({ identifier, time }, roundId);
+    const retrievedEncryptedMessage = await voting.getMessage(account1, topicHash);
+
+    // Check that the stored message is correct.
+    assert.equal(encryptedMessage, retrievedEncryptedMessage);
+
+    await moveToNextPhase(voting);
+
+    const decryptedMessage = await decryptMessage(privateKey, retrievedEncryptedMessage);
+    const retrievedVote = JSON.parse(decryptedMessage);
+
+    assert(await didContractThrow(voting.revealVote(identifier, time, getRandomSignedInt(), getRandomUnsignedInt())));
+    await voting.revealVote(identifier, time, retrievedVote.price, retrievedVote.salt);
+  });
+
+  it("Commit and persist multiple messages", async function() {
```

Comment:
nit: can we change the title to note that this test rewrites a single commit rather than commits twice?